File: CSee/Source.cpp
#include "Source.h"


Source::Source(std::string& name, bool showRaw, std::string& arName, std::string& cvName)
{
	this->name = name;
	this->arName = arName;
	this->cvName = cvName;
	this->_showRaw = showRaw;

	if (this->showRaw()){
		connectionNames.push_back(std::string(name + ".raw"));
		channelNames.push_back(std::string(name + ".raw"));
	}
	if (this->hasAR()){
		connectionNames.push_back(std::string(name + "." + arName));
		channelNames.push_back(std::string(name + "." + arName));
	}
	if (this->hasCV()){
		connectionNames.push_back(std::string(name + "." + cvName));
	}

}

Source::~Source()
{
}

std::string Source::getName(){
	return name;
}

std::string Source::getARName(){
	return arName;
}

std::string Source::getCVName(){
	return cvName;
}

bool Source::showRaw(){
	return _showRaw;
}
bool Source::hasAR(){
	return !arName.empty();
}
bool Source::hasCV(){
	return !cvName.empty();
}

int Source::countConnections(){
	return connectionNames.size();
}

int Source::countChannels(){
	return channelNames.size();
}

std::vector< std::string> Source::getChannelNames(){
	return channelNames;
}
std::vector<std::string> Source::getConnectionNames(){
	return connectionNames;
}

File: MSee/src/team4141/robotvision/msee/Source.java
package team4141.robotvision.msee;

import java.util.Hashtable;

public abstract class Source {
	
	protected Hashtable<String,Setting> settings;
	protected String name;
	protected String description;
	protected String id;
	protected Boolean showRaw;
	protected String arName;
	protected String cvName;

	public Source(String id, String description){
		this.id = id;
		this.description = description;
		this.settings = new Hashtable<String,Setting>();
	}
	@Override
	public String toString() {
		return getId();
	}
	public void add(Setting setting) {
		settings.put(setting.getName(), setting);
	}
	public Setting get(String settingName) {
		if(!settings.containsKey(settingName)) return null;
		return settings.get(settingName);
	}
	public String getName(){
		return name;
	}
	public String getARName(){
		return arName;
	}
	public String getCVName(){
		return cvName;
	}
	public String getId() {
		return id;
	}	
	public String getDescription(){
		return description;
	}
	public void setName(String name){
		this.name = name;
	}
	public void setARName(String arName){
		this.arName = arName;
	}
	public void setCVName(String cvName){
		this.cvName = cvName;
	}
	public Boolean showRaw() {
		return showRaw;
	}		
	public void setShowRaw(boolean showRaw) {
		this.showRaw = showRaw;
	}		
}
